app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt

```diff
@@ -244,6 +275,15 @@ class CtaViewModel @Inject constructor(
     @WorkerThread
     private suspend fun canShowDaxIntroCta(): Boolean = daxOnboardingActive() && !daxDialogIntroShown() && !settingsDataStore.hideTips
 
+    @WorkerThread
+    private suspend fun canShowDaxFireproofCta(): Boolean = daxOnboardingActive() &&
+        !daxDialogEndShown() &&
+        !daxDialogFireproofShown() &&
+        daxDialogIntroShown() &&
+        !settingsDataStore.hideTips &&
+        userEventsStore.getUserEvent(UserEventKey.PROMOTED_FIRE_BUTTON_CANCELLED) == null &&
+        variantManager.isFireproofExperimentEnabled()
```

Comment:
nit: could check the variant first and avoid the other checks

Reply:
That check is now the first one 👍